ee/codegen/src/generators/vellum-variable-value.ts

```diff
@@ -283,32 +283,52 @@ class SearchResultsVellumValue extends AstNode {
         }),
         python.methodArgument({
           name: "document",
-          value: python.reference({
-            name: "document", // Assuming document is already instantiated
-            modulePath: VELLUM_CLIENT_MODULE_PATH,
+          value: python.instantiateClass({
```

Comment:
@noanflaherty Could you take another look here to make sure this looks right for inheritance and document instatiation?